staging/src/k8s.io/legacy-cloud-providers/azure/azure_blobDiskController.go

```diff
@@ -300,10 +300,9 @@ func (c *BlobDiskController) getStorageAccountKey(SAName string) (string, error)
 				klog.Warningf("azureDisk - account %s was not cached while getting keys", SAName)
 				return *v.Value, nil
 			}
+			c.accounts[SAName].key = *v.Value
```

Comment:
why move to the inner scope?

Reply:
It make sense to move into the if statement, the `inner if` is checking whether the c.accounts[SAName] is empty or not, if it is empty, this func just return the result, else, this func will cache the result into c.accounts[SAName].key and return the same result.
If it just follows the outer, this will break the for loop in one step and return result is strange.
What do you think?

Reply:
this seems to be dramatically changing behavior. that said, the existing code is confusing as well:
* line 298 checks if v.Value is non-nil, but `c.accounts[SAName].key = *v.Value` unconditionally dereferenced it
* `if _, ok := c.accounts[SAName]; !ok {` checks if there's an entry in accounts under SAName, but `c.accounts[SAName].key = *v.Value` unconditionally assumed there is

can we split this change out of a "fix static checks" PR into a bug fix PR?

Reply:
Sure, I will do this.